boards/msp_exp432p401r/src/main.rs

```diff
@@ -0,0 +1,165 @@
+#![no_std]
+// Disable this attribute when documenting, as a workaround for
+// https://github.com/rust-lang/rust/issues/62184.
+#![cfg_attr(not(doc), no_main)]
+// #![deny(missing_docs)]
+
+use kernel::capabilities;
+use kernel::common::dynamic_deferred_call::DynamicDeferredCall;
+use kernel::common::dynamic_deferred_call::DynamicDeferredCallClientState;
+use kernel::component::Component;
+use kernel::hil::watchdog::Watchdog;
+use kernel::Platform;
+use kernel::{create_capability, debug, static_init};
+
+pub mod io;
+
+/// Number of concurrent processes this platform supports.
+const NUM_PROCS: usize = 4;
+
+/// Actual memory for holding the active process structures.
+static mut PROCESSES: [Option<&'static dyn kernel::procs::ProcessType>; NUM_PROCS] =
+    [None; NUM_PROCS];
+
+/// Static reference to chip for panic dumps.
+static mut CHIP: Option<&'static msp432::chip::Msp432> = None;
+
+/// How should the kernel respond when a process faults.
+const FAULT_RESPONSE: kernel::procs::FaultResponse = kernel::procs::FaultResponse::Panic;
+
+/// RAM to be shared by all application processes.
+#[link_section = ".app_memory"]
+static mut APP_MEMORY: [u8; 32768] = [0; 32768];
```

Comment:
Please update the `APP_MEMORY` to the new model in #1962 -- sorry these came somewhat concurrently :/

Reply:
Sure, I will convert it to the new model